### PR TITLE
Issue #KN-239 feat: Knowlg player app for Video player and library migration changes 

### PR DESCRIPTION
**Pull request by santoshilimi**
Player apk: https://drive.google.com/file/d/1k1ZhF-0uiIKAkFa9xl2m_gXyIU2ymO0A/view?usp=sharing

### CODE REVIEW

File: knowlg-app/package.json [hunk 1]
```diff
@@ -13,40 +13,49 @@
   },
   "private": true,
   "dependencies": {
-    "@angular/common": "^14.0.0",
```
Comment (HarishGangula) [inline knowlg-app/package.json:16]: Why angular is downgraded

File: knowlg-app/src/app/app.module.ts [hunk 2]
```diff
@@ -8,11 +8,17 @@ import { AppComponent } from './app.component';
 import { AppRoutingModule } from './app-routing.module';
 import { SunbirdPdfPlayerModule } from '@project-sunbird/sunbird-pdf-player-v9';
 import { SunbirdEpubPlayerModule } from '@project-sunbird/sunbird-epub-player-v9';
-
+import { QuestionCursor } from '@project-sunbird/sunbird-quml-player-v9';
```
Comment (HarishGangula) [inline knowlg-app/src/app/app.module.ts]: Question cursor is optional cross it

File: package.json [hunk 3]
```diff
@@ -83,9 +82,8 @@
     "karma-chrome-launcher": "~3.1.0",
     "karma-coverage": "^2.2.0",
     "karma-jasmine": "~4.0.0",
-    "ng-packagr": "^9.0.0",
     "karma-jasmine-html-reporter": "^1.5.0",
-    "karma-coverage-istanbul-reporter": "~2.1.0",
+    "ng-packagr": "^12.1.1",
```
Comment (HarishGangula) [inline package.json:86]: This changes are regarding

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (santoshilimi) [reply package.json:86]: Before library was in angular 9 and portal application also was in 9 version but after migration library was not supporting so did these changes for compatible.

File: knowlg-app/package.json (hunk 1 above)
Comment (santoshilimi) [reply knowlg-app/package.json:16]: Before library was in angular 9 and portal application also was in 9 version but after migration library was not supporting so did these changes for compatible.

File: knowlg-app/src/app/app.module.ts (hunk 2 above)
Comment (santoshilimi) [reply knowlg-app/src/app/app.module.ts]: As i am  getting No provider for QuestionCursor , because if it i have added.

**Review by HarishGangula — APPROVED** [review]

File: knowlg-app/angular.json [hunk 4]
```diff
@@ -25,21 +25,42 @@
...
-              }
+              },
+              {
+                "glob": "**/*.*",
+                "input": "./node_modules/@project-sunbird/sunbird-video-player-v9/lib/assets/",
+                "output": "/assets/"
+              },
+            {
+                  "glob": "**/*",
+                  "input": "node_modules/@project-sunbird/sunbird-quml-player-v9/lib/assets/",
```
Comment (vinukumar-vs) [inline knowlg-app/angular.json]: Why are we adding QUML player here? 
We dont need QUML player to verify

File: knowlg-app/package.json [hunk 5]
```diff
@@ -13,40 +13,49 @@
...
     "@capacitor/haptics": "4.0.1",
     "@capacitor/keyboard": "4.0.1",
     "@capacitor/status-bar": "4.0.1",
     "@ionic/angular": "^6.1.9",
+    "@ionic/cli": "^6.20.1",
     "@project-sunbird/client-services": "^4.9.1",
     "@project-sunbird/sb-styles": "0.0.9",
     "@project-sunbird/sunbird-epub-player-v9": "^4.8.2",
     "@project-sunbird/sunbird-pdf-player-v9": "^5.1.0",
+    "@project-sunbird/sunbird-quml-player-v9": "^5.1.3",
```
Comment (vinukumar-vs) [inline knowlg-app/package.json]: Is the QUML player is a dependecy for Video player?

File: knowlg-app/src/app/home/services/data.ts [hunk 6]
```diff
@@ -337,7 +337,110 @@ export const  data = {
         }
       }
       ],
-      video: [],
+      video: [
+        {
+          "interceptionPoints": "{'items':[{'type':'QuestionSet','interceptionPoint':50,'identifier':'do_213272808198291456121'},{'type':'QuestionSet','interceptionPoint':90,'identifier':'do_213272808198291456121'},{'type':'QuestionSet','interceptionPoint':120,'identifier':'do_213272808198291456121'}]}",
+          "interceptionType": "Timestamp",
+          "compatibilityLevel": 2,
```
Comment (vinukumar-vs) [inline knowlg-app/src/app/home/services/data.ts:483]: We should not test for question-set data

File: knowlg-app/src/app/app.module.ts [hunk 7]
```diff
@@ -8,11 +8,17 @@ import { AppComponent } from './app.component';
...
+import { QuestionCursorImplementationService } from './question-cursor-implementation.service';
+import { SunbirdVideoPlayerModule } from '@project-sunbird/sunbird-video-player-v9';
 @NgModule({
   declarations: [AppComponent],
-  imports: [BrowserModule, IonicModule.forRoot(), AppRoutingModule, SunbirdPdfPlayerModule, SunbirdEpubPlayerModule],
-  providers: [{ provide: RouteReuseStrategy, useClass: IonicRouteStrategy }],
+  imports: [BrowserModule, IonicModule.forRoot(), AppRoutingModule, SunbirdPdfPlayerModule,
+     SunbirdEpubPlayerModule, SunbirdVideoPlayerModule],
+  providers: [{ provide: RouteReuseStrategy, useClass: IonicRouteStrategy },
+   { provide: QuestionCursor,
```
Comment (vinukumar-vs) [inline knowlg-app/src/app/app.module.ts]: Why do we have this question related implementation here?

File: package.json (hunk 3 above)
Comment (vinukumar-vs) [inline package.json:86]: Why did you removed Karma coverage report? THis is used by sonar cloub right?

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (santoshilimi) [reply package.json:86]: we have karma-coverage so karma-coverage-istanbul-reporter is not required

File: knowlg-app/angular.json (hunk 4 above)
Comment (santoshilimi) [reply knowlg-app/angular.json]: Removed

File: knowlg-app/src/app/home/services/data.ts (hunk 6 above)
Comment (santoshilimi) [reply knowlg-app/src/app/home/services/data.ts:483]: Removed

File: knowlg-app/package.json (hunk 5 above)
Comment (santoshilimi) [reply knowlg-app/package.json]: Yes QUML player is dependency but checking on it to remove

File: knowlg-app/src/app/app.module.ts (hunk 7 above)
Comment (santoshilimi) [reply knowlg-app/src/app/app.module.ts]: Yes QUML player is dependency but checking on it to remove

**Review by HarishGangula — APPROVED** [review]

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**